sqlmesh/core/context.py

```diff
@@ -2148,6 +2157,36 @@ def _load_requirements(self, path: Path) -> None:
                         )
                     self._requirements[dep] = ver
 
+    def _load_multiple_gateways(self) -> None:
+        # Validate there are multiple gateways and the default is duckdb
+        if (
+            len(self.config.gateways) < 2
+            or not isinstance(self._connection_config, DuckDBConnectionConfig)
+            or not self._connection_config.database
+            or self._connection_config.catalogs
+        ):
+            return
+
+        # Set the attach options for the main and additional connection gateways (Currently supported DuckDB and PostgreSQL)
+        self._connection_config.catalogs = {
+            self._connection_config.database.split(".")[0]: self._connection_config.database
+        }
+        for gateway_name, config in self.config.gateways.items():
+            if isinstance(config.connection, PostgresConnectionConfig):
+                self._connection_config.catalogs[gateway_name] = DuckDBAttachOptions(
+                    type=config.connection.type_, path=config.connection._libpq_string
+                )
+            elif (
+                isinstance(config.connection, DuckDBConnectionConfig)
+                and config.connection.database
+                and config.connection.database != self._connection_config.database
+            ):
+                self._connection_config.catalogs[config.connection.database.split(".")[0]] = (
```

Comment:
Are you trying to parse the catalog out of here? It might be best to use a sqlglot parsing function, i'm thinking `exp.to_table()` but @georgesittas will know the most appropriate one

Reply:
Yes it was to get for example from "main.db" the database name so "main", but maybe I could use the gateway name there instead directly assuming it is the same as the catalog name

Reply:
> but maybe I could use the gateway name there instead directly assuming it is the same as the catalog name

This assumption looks incorrect, catalogs and gateway names are orthogonal concepts. See `stem` suggestion regarding the string splitting bit.